### PR TITLE
Add versions page and nav version

### PR DESCRIPTION
**Pull request by rickhanlonii**
## Overview

For React 19, we need to remove all of the removed APIs from the docs. This poses a problem for users on older versions, who still need access to docs for removed APIs before they've upgraded to 19.

In this PR, I add a `versions` page that:
- Lists the pinned links to old docs versions
- Explains that we maintain old docs per major version
- Lists the changelog and relevant blog posts and talks per major release
- Shows the initial commit and open source release talk from 2013
- Adds redirects
  - 17.react.dev -> https://17.reactjs.org/
  - 16.react.dev -> https://17.reactjs.org/
    - Note: the 16 docs are broken, so for now I'm redirecting to 17
  - 15.react.dev -> https://react-legacy.netlify.app/
  
## Screenshot
  
Screenshot 2024-04-29 at 6 35 45 PM

## Initial Commit
This is fun

Screenshot 2024-04-29 at 6 35 52 PM

## Testing

Tested on light mode and dark mode, on desktop and mobile.

### CODE REVIEW

File: src/content/versions.md [hunk 1]
```diff
@@ -0,0 +1,282 @@
...
+- [React v0.14 Release Candidate](https://legacy.reactjs.org/blog/2015/09/10/react-v0.14-rc1.html)
+- [React v0.14 Beta 1](https://legacy.reactjs.org/blog/2015/07/03/react-v0.14-beta-1.html)
+- [New React Developer Tools](https://legacy.reactjs.org/blog/2015/09/02/new-react-developer-tools.html)
+- [New React Devtools Beta](https://legacy.reactjs.org/blog/2015/08/03/new-react-devtools-beta.html)
+- [React v0.14.1](https://legacy.reactjs.org/blog/2015/10/28/react-v0.14.1.html)
+- [React v0.14.2](https://legacy.reactjs.org/blog/2015/11/02/react-v0.14.2.html)
+- [React v0.14.3](https://legacy.reactjs.org/blog/2015/11/18/react-v0.14.3.html)
+- [React v0.14.4](https://legacy.reactjs.org/blog/2015/12/29/react-v0.14.4.html)
+- [React v0.14.8](https://legacy.reactjs.org/blog/2016/03/29/react-v0.14.8.html)
+-
```
Comment (noahlemen) [inline src/content/versions.md:179]: 👀 empty bullet

**Review by noahlemen — COMMENTED** [review]
nice! should the "previous versions" links be in a bulleted list, for consistent formatting with the rest of the page?

**Comment by rickhanlonii** [comment]
Talked with the team, I think we will want to keep the "current" version at `react.dev` as 18 + canary for now, so I'll add a "Future Versions" section to the page with a link to a React 19 branch.

Then for https://github.com/reactjs/react.dev/pull/6811 we'll land it on the 19 branch and i'll manually keep changes in sync between the two until 19 stable is out.

File: src/content/versions.md [hunk 2]
```diff
@@ -0,0 +1,282 @@
...
+
+</Intro>
+
+We aim to keep the docs updated within major versions, and do not publish versions for each minor or patch version. When a new major is released, we archive the docs for the previous version as `x.react.dev`. See our [versioning policy](/community/versioning-policy) for more info.
+
+You can find an archive of previous major versions below.
+
+## Previous doc versions {/*previous-doc-versions*/}
+
+#### [18.react.dev](https//18.react.dev) {/*docs-18*/}
```
Comment (kassens) [inline src/content/versions.md:17]:
```suggestion
#### [18.react.dev](https://18.react.dev) {/*docs-18*/}
```

Comment (rickhanlonii) [reply src/content/versions.md]: Fixed

**Pull request merged**